src/theia/sfm/feature_extractor.cc

```diff
@@ -102,34 +127,15 @@ bool FeatureExtractor::ExtractFeatures(
     std::vector<Keypoint>* keypoints,
     std::vector<Eigen::VectorXf>* descriptors) {
   std::unique_ptr<FloatImage> image(new FloatImage(filename));
-
-  // We create these variable here instead of upon the construction of the
-  // object so that they can be thread-safe. We *should* be able to use the
-  // static thread_local keywords, but apparently Mac OS-X's version of clang
-  // does not actually support it!
-  //
-  // TODO(cmsweeney): Change this so that each thread in the threadpool receives
-  // exactly one object.
-  std::unique_ptr<DescriptorExtractor> descriptor_extractor =
-      CreateDescriptorExtractor(options_.descriptor_extractor_type,
-                                options_.feature_density);
-
-  // Exit if the descriptor extraction fails.
-  if (!descriptor_extractor->DetectAndExtractDescriptors(*image,
-                                                         keypoints,
-                                                         descriptors)) {
+  if (!ExtractFeaturesFromImage(*image, keypoints, descriptors)) {
     LOG(ERROR) << "Could not extract descriptors in image " << filename;
     return false;
   }
-
-  if (keypoints->size() > options_.max_num_features) {
-    keypoints->resize(options_.max_num_features);
-    descriptors->resize(options_.max_num_features);
+  else {
```

Comment:
In theia, we use brackets on the same line for if/else like this:

if (something) {
  DoTheThing();
} else {
  DoTheOtherThing();
}

Github isn't showing it in the comments, but we also use a 2-space indentation.